## 1. Objective

The objective of this notebook is to load the Telco Customer Churn dataset and carry out a first inspection. This includes:

- Reading the CSV file into a DataFrame
- Verifying dimensions, datatypes, and structure
- Identifying possible data quality issues (e.g., nulls, types, duplicates)
- Saving a working copy in `data/processed/` for use in upcoming notebooks


2. Loading the Dataset

In [2]:
import pandas as pd
import numpy as np
import os

# Set display options
pd.set_option('display.max_columns', None)

# Path to dataset
DATA_PATH = "../data/raw/WA_Fn-UseC_-Telco-Customer-Churn.csv"

# Load data
try:
    df = pd.read_csv(DATA_PATH)
    print("✅ Dataset loaded successfully.")
except FileNotFoundError:
    print("❌ Dataset not found. Please check the path:", DATA_PATH)

# Preview
df.head()


✅ Dataset loaded successfully.


,customerID,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,OnlineBackup,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,MonthlyCharges,TotalCharges,Churn
0,7590-VHVEG,Female,0,Yes,No,1,No,No phone service,DSL,No,Yes,No,No,No,No,Month-to-month,Yes,Electronic check,29.85,29.85,No
1,5575-GNVDE,Male,0,No,No,34,Yes,No,DSL,Yes,No,Yes,No,No,No,One year,No,Mailed check,56.95,1889.5,No
2,3668-QPYBK,Male,0,No,No,2,Yes,No,DSL,Yes,Yes,No,No,No,No,Month-to-month,Yes,Mailed check,53.85,108.15,Yes
3,7795-CFOCW,Male,0,No,No,45,No,No phone service,DSL,Yes,No,Yes,Yes,No,No,One year,No,Bank transfer (automatic),42.30,1840.75,No
4,9237-HQITU,Female,0,No,No,2,Yes,No,Fiber optic,No,No,No,No,No,No,Month-to-month,Yes,Electronic check,70.70,151.65,Yes


3. Initial Data Inspection

In [3]:
# Shape of the dataset
print(f"📊 Rows: {df.shape[0]}, Columns: {df.shape[1]}")

# Column data types
df.dtypes

# Null values per column
print("🕳️ Missing values per column:\n")
print(df.isnull().sum())

# Basic descriptive statistics
df.describe(include='all').T

📊 Rows: 7043, Columns: 21
🕳️ Missing values per column:

customerID          0
gender              0
SeniorCitizen       0
Partner             0
Dependents          0
tenure              0
PhoneService        0
MultipleLines       0
InternetService     0
OnlineSecurity      0
OnlineBackup        0
DeviceProtection    0
TechSupport         0
StreamingTV         0
StreamingMovies     0
Contract            0
PaperlessBilling    0
PaymentMethod       0
MonthlyCharges      0
TotalCharges        0
Churn               0
dtype: int64


,count,unique,top,freq,mean,std,min,25%,50%,75%,max
customerID,7043,7043,7590-VHVEG,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN
gender,7043,2,Male,3555,NaN,NaN,NaN,NaN,NaN,NaN,NaN
SeniorCitizen,7043.0,NaN,NaN,NaN,0.162147,0.368612,0.0,0.0,0.0,0.0,1.0
Partner,7043,2,No,3641,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Dependents,7043,2,No,4933,NaN,NaN,NaN,NaN,NaN,NaN,NaN
tenure,7043.0,NaN,NaN,NaN,32.371149,24.559481,0.0,9.0,29.0,55.0,72.0
PhoneService,7043,2,Yes,6361,NaN,NaN,NaN,NaN,NaN,NaN,NaN
MultipleLines,7043,3,No,3390,NaN,NaN,NaN,NaN,NaN,NaN,NaN
InternetService,7043,3,Fiber optic,3096,NaN,NaN,NaN,NaN,NaN,NaN,NaN
OnlineSecurity,7043,3,No,3498,NaN,NaN,NaN,NaN,NaN,NaN,NaN


4. Save Working

In [4]:
# Save a working version in /data/processed
output_path = "../data/processed/telco_clean_base.csv"
df.to_csv(output_path, index=False)
print(f"💾 Data saved to: {output_path}")

💾 Data saved to: ../data/processed/telco_clean_base.csv


## Summary

- The dataset was successfully loaded with 7043 rows and 21 columns.
- We inspected the data types, basic statistics, and missing values.
- A base working version of the dataset was saved for future processing and modeling steps.

Next Notebook `02_eda.ipynb`: Exploratory Data Analysis (EDA).